Introduce new `SSLContext` API & escalate deprecations. (#3319)
encode/httpx: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

## Unreleased

* Added `httpx.SSLContext` class and `ssl_context` argument. (#3022)

> * Removed `cert` and `verify` arguments, you should use the `ssl_context=...` instead. (#3022)

Author: * The `verify` and `cert` arguments continue to function, but are deprecated and will raise warnings. Use the `ssl_context = httpx.SSLContext()` API instead. (#3022, #3335)

> * Removed `cert` and `verify` arguments, you should use the `ssl_context=...` instead. (#3022)

Author: ```suggestion
* The `verify` and `cert` arguments continue to function, but are deprecated and will raise warnings. Use the `ssl_context = httpx.SSLContext()` API instead. (#3022, #3335)
```

* The deprecated `proxies` argument has now been removed.
* The deprecated `app` argument has now been removed.
* The `URL.raw` property has now been removed.

> * The `sniffio` project dependency has now been removed.

Author: ```suggestion
```

Dropping this note, since it's internal.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).

> ## Unreleased

Author: ```suggestion
## Version 0.28.0
```

> Version <...> introduces an `httpx.SSLContext()` class and `ssl_context` parameter.

Author: ```suggestion
Version 0.28.0 introduces an `httpx.SSLContext()` class and `ssl_context` parameter.
```

> ## 0.28.0 (28th October, 2024)
> 
> Version 0.28.0 introduces an `httpx.SSLContext()` class and `ssl_context` parameter.

Author: ```suggestion
## Unreleased

Version <...> introduces an `httpx.SSLContext()` class and `ssl_context` parameter.
```
